datalog/analysis/dead-code.dl

```diff
@@ -140,13 +140,18 @@ ReachableCode(n) :-
 
 
 .decl EffectfulFunction(f:Function)
+.output EffectfulFunction(delimiter=",")
 
+// QUESTION: Why do we need f to have parameters?
 EffectfulFunction(f)
   :- FunctionParameter(f,_,_)
    , CodeNameInst(f,n)
    , Effectful(n).
 
+// QUESTION: CallArgument(n,_,_) means that it is not a nullary function? Why do we need this restriction?
 Effectful(n) :- External(f,1), Call(n, f), CallArgument(n,_,_).
+// QUESTION: Is Update really effectful? Didn't we settle on a conclusion that Updates can only be inside evals
```

Comment:
This is faithful to the Boq PhD, but since then we changed the semantics of "effectful". In the original PhD everything that returned `()` was effectful, but now only externals that are marked effectful are effectful.